src/factor.h

```diff
@@ -62,11 +73,19 @@ class CompactFactor {
    */
   inline double potential(const VariableInFactor vifs[],
                           const variable_value_t var_values[],
-                          const variable_id_t vid,
-                          const variable_value_t proposal) const {
+                          const variable_id_t vid = -1,
+                          const variable_value_t proposal = -1,
```

Comment:
Changed values. There are also a couple calling sites that set vid and proposal but not feature_value, so these two are probably not more optional than feature_value...